lib/MusicBrainz/Server/Edit/Relationship/Delete.pm

```diff
@@ -268,12 +268,12 @@ before restore => sub {
                 type => {
                     root => {
                         id => $_->{root_id},
-                        gid => $_->{root_gid},
+                        $_->{root_gid} ? (gid => $_->{root_gid}) : (),
                         name => $_->{root_name},
                     },
-                    id => $_->{id},
-                    gid => $_->{gid},
-                    name => $_->{name},
+                    id => ($_->{id} // $_->{root_id}),
+                    $_->{gid} ? (gid => $_->{gid}) : (),
```

Comment:
Note: there doesn't seem to ever have been a case where `root_gid` was present and not `gid`, since they were added at the same time. So there's no fallback here. See https://github.com/mwiencek/musicbrainz-server/commit/551a9753c2c12d2174577c36710588a95ce0d5c0